packages/modal/src/Modal.js

```diff
@@ -1,6 +1,72 @@
-import { Modal } from "hig-react";
-import "./modal.scss";
+import React, { Component } from "react";
+import PropTypes from "prop-types";
 
-Modal.displayName = "Modal";
+import ModalPresenter from "./presenters/ModalPresenter";
+import ModalBehavior from "./behaviors/ModalBehavior";
 
-export default Modal;
+export default class Modal extends Component {
+  static propTypes = {
+    /**
+     * Supports adding any dom content to the body of the modal
+     */
+    children: PropTypes.node,
+    /**
+     * Supports adding any dom content to the header of the modal
+     */
+    headerChildren: PropTypes.node,
+    /**
+     * Triggers when you click the close button
+     */
+    onCloseClick: PropTypes.func,
+    /**
+     * Triggers when you click the overlay behind the modal
+     */
+    onOverlayClick: PropTypes.func,
+    /**
+     * Modal is visible when true
+     */
+    open: PropTypes.bool,
+    /**
+     * Style of the modal shell
+     */
+    style: PropTypes.string,
```

Comment:
I understand it was called `style` before, but we really should change since `style` has a very specific mean in React. We've been using `type` in most of our components.

Reply:
Good point about React `style`

Two things though:

1. I can't stand the word `type`, might as well use `blah` but again, I see the value in consistency so whatever
2. We should try and front load more API talk

Reply:
Updated in 275762f3

Reply:
> I can't stand the word type, might as well use blah but again, I see the value in consistency so whatever

We should be using the word `variant` since that's what the designers use in their documentation, but we've been sticking with `type` for consistency during the migration. In `v1` I'd like to see `variant` used across the board in every component.

Reply:
👍